### PR TITLE
Remove ops from Layer Creator/ Node Converter - part 5

### PR DESCRIPTION
**Pull request by blesniewski**
Num | Removed op name
-- | --
1 | AvgPool
2 | BinaryConvolution
3 | Broadcast
4 | CTCGreedyDecoder
5 | StridedSlice
6 | ConvolutionBackpropData

43472

### CODE REVIEW

**Review by sdurawa — APPROVED** [review]

File: inference-engine/src/legacy_api/src/convert_function_to_cnn_network.cpp [hunk 1]
```diff
@@ -1291,6 +1291,15 @@ InferenceEngine::details::CNNLayerCreator::CNNLayerCreator(const std::shared_ptr
         }
         return res;
     });
+
+        addSpecificCreator({"CTCGreedyDecoder"}, [](const std::shared_ptr<::ngraph::Node>& node,
```
Comment (ilyachur) [inline inference-engine/src/legacy_api/src/convert_function_to_cnn_network.cpp:1295]: nitpick: Incorrect indent

File: ngraph/core/src/op/broadcast.cpp [hunk 2]
```diff
@@ -269,8 +269,31 @@ op::v1::Broadcast::Broadcast(const Output<Node>& arg,
...
+    {
+        m_broadcast_spec.m_type = AutoBroadcastType::NUMPY;
+    }
+
+    // Mocking axes_mapping input for cases that don't require it
+    if (m_broadcast_spec.m_type == AutoBroadcastType::NUMPY)
+    {
+        if (get_input_size() < 3)
+        {
+            auto output = op::v0::Constant::create(element::u8, Shape{}, {0})->output(0);
```
Comment (ilyachur) [inline ngraph/core/src/op/broadcast.cpp:284]: Why do we need this code?

File: ngraph/core/src/op/broadcast.cpp [hunk 3]
```diff
@@ -269,8 +269,31 @@ op::v1::Broadcast::Broadcast(const Output<Node>& arg,
...
+    // has 2 inputs its always NUMPY mode
+    if (m_broadcast_spec.m_type == AutoBroadcastType::NONE && get_input_size() < 3)
+    {
+        m_broadcast_spec.m_type = AutoBroadcastType::NUMPY;
+    }
+
+    // Mocking axes_mapping input for cases that don't require it
+    if (m_broadcast_spec.m_type == AutoBroadcastType::NUMPY)
+    {
+        if (get_input_size() < 3)
```
Comment (ilyachur) [inline ngraph/core/src/op/broadcast.cpp:282]: I think we can merge these two `if`s

File: ngraph/core/src/op/broadcast.cpp (hunk 2 above)
Comment (blesniewski) [reply ngraph/core/src/op/broadcast.cpp:284]: Broadcast op has a default constructor, and two other ones (one that has 2 and one has 3 inputs). 
If we're creating this op through the 2-input constructor, we're also doing this mocking inside the constructor.
But, in this case we're dealing with creating broadcast op through the default constructor, and we have two inputs (so the third input should be mocked but is not), so we're mocking it, you can take look at the v1::broadcast op's constructors to see what I mean.

**Review by ilyachur — APPROVED** [review]

**Pull request merged**